pkg/resources/sources/cpu.go

```diff
@@ -31,13 +37,16 @@ func (c *CPU) GetTotalResources(ctx context.Context) (core.ResourceList, error)
 		hostCPUSum += int64(v.Cores)
 	}
 
-	resources := core.ResourceList{core.ResourceCPU: *resource.NewScaledQuantity(hostCPUSum, resource.Kilo)}
+	cpuQuantity := float64(hostCPUSum) * c.OvercommitVCPU
+	resources := core.ResourceList{
```

Comment:
why did you change this operation and you didn't used NewScaledQuantity?

Reply:
The change was made because NewScaledQuantity doesn't handle floating-point scaling well. NewScaledQuantity expects an integer as the first argument and a resource.Scale as the second argument. Since c.OvercommitVCPU can be a float, we needed to perform the scaling manually by converting the result to an integer. This manual scaling ensures that fractional values like 1.5 or 0.5 are correctly converted to their corresponding values in kilo. Using NewScaledQuantity directly wouldn't have allowed us to handle such fractional values correctly.

Reply:
but you don't use float and you retype it to int64, so i am confuse.

Reply:
The reason for manually scaling the CPU quantity and converting it to an integer (int64) is to ensure that fractional values, such as 1.5 or 0.5, are correctly converted to their corresponding values in kilo. Even though the c.OvercommitVCPU value may be a float, the final result needs to be an integer representing the CPU quantity in kilo.

The NewScaledQuantity function expects an integer as its first argument, which is why we need to perform the scaling manually. By converting the result to an integer, we ensure that the CPU quantity is correctly represented in kilo, even if the c.OvercommitVCPU value is a float.

Using NewScaledQuantity directly wouldn't allow us to handle such fractional values correctly, as it expects an integer value for scaling. Therefore, the manual scaling and conversion to an integer are necessary in this case.

Reply:
As discussed offline, using `NewScaledQuantity`